File: leetcode/1701/averageWaitingTime.swift
class Solution {
    func averageWaitingTime(_ customers: [[Int]]) -> Double {
        var wait = Int(0)
        var total = Int(0)

        for customer in customers{
            if total < customer[0]{
                total = customer[0] + customer[1]
                wait += total - customer[0]
            }
            else{
                wait += total-customer[0] + customer[1]
                total += customer[1]
            }
        }

        return Double(wait) / Double(customers.count)
    }
}

File: leetcode/1701/averageWaitingTime.py
class Solution:
    def averageWaitingTime(self, customers: List[List[int]]) -> float:
        wait = 0
        total = 0
        for (i,j) in customers:
            if total < i:
                total = i
                total += j
                wait += total - i
            else:
                wait += total-i + j
                total += j

        return wait/len(customers)

